ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/error/RedirectSSOErrorTest.kt

```diff
@@ -17,181 +17,171 @@ import com.jayway.restassured.RestAssured
+                    createDefaultAuthnRequest(SamlProtocol.Binding.HTTP_REDIRECT)
                 val encodedRequest = encodeAuthnRequest(authnRequest)
                 val queryParams = SimpleSign().signUriString(
-                        SAML_REQUEST,
-                        encodedRequest,
-                        null)
+                    SAML_REQUEST,
+                    encodedRequest,
+                    null)
                 val qpSamlReq = queryParams[SAML_REQUEST]
                 queryParams.set(SAML_REQUEST, qpSamlReq?.substring(0, qpSamlReq.length / 2))
-
-                // Get response from AuthnRequest
                 val response = sendRedirectAuthnRequest(queryParams)
 
-                val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
+                if (!isLenient || !BindingVerifier.isErrorHttpStatusCode(response.statusCode)) {
+                    val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
 
-                CoreVerifier.verifyErrorStatusCode(samlResponseDom,
+                    CoreVerifier.verifyErrorStatusCode(samlResponseDom,
                         samlErrorCode = SAMLBindings_3_4_3_a,
                         expectedStatusCode = REQUESTER)
+                }
             } catch (e: SAMLComplianceException) {
                 throw SAMLComplianceException.recreateExceptionWithErrorMessage(e)
             }
         }
 
-        "Empty Redirect AuthnRequest Test" {
+        "Profiles 4.1.4.1: Empty Redirect AuthnRequest Test" {
             try {
-                Log.debugWithSupplier { "Empty Redirect AuthnRequest Test" }
+                Log.debugWithSupplier { "Profiles 4.1.4.1: Empty Redirect AuthnRequest Test" }
                 val authnRequest = AuthnRequestBuilder().buildObject()
                 val encodedRequest = encodeAuthnRequest(authnRequest)
                 val queryParams = mapOf(SAML_REQUEST to encodedRequest)
-
                 val response = sendRedirectAuthnRequest(queryParams)
 
-                val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
+                if (!isLenient || !BindingVerifier.isErrorHttpStatusCode(response.statusCode)) {
+                    val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
 
-                CoreVerifier.verifyErrorStatusCode(samlResponseDom,
+                    CoreVerifier.verifyErrorStatusCode(samlResponseDom,
                         samlErrorCode = SAMLProfiles_4_1_4_1_a,
                         expectedStatusCode = REQUESTER)
-                ProfilesVerifier(samlResponseDom).verifyErrorResponseAssertion()
+                    ProfilesVerifier(samlResponseDom).verifyErrorResponseAssertion()
+                }
             } catch (e: SAMLComplianceException) {
                 throw SAMLComplianceException.recreateExceptionWithErrorMessage(e)
             }
         }
 
-        "Redirect AuthnRequest With Empty Subject Test" {
```

Comment:
❓ Why was the `Empty Subject Test` removed/changed?

Reply:
☝️